lib/tasks/scripts/block_level_sync.rb

```diff
@@ -0,0 +1,74 @@
+class BlockLevelSync
+  class << self
+    def disable(user_ids)
+      new.toggle_state(user_ids, false)
+    end
+
+    def enable(user_ids)
+      new.toggle_state(user_ids, true)
+    end
+
+    def bump(percentage)
+      new.bump(percentage)
+    end
+  end
+
+  def toggle_state(user_ids, enable)
+    user_ids.each do |user_id|
+      user = User.find_by(id: user_id)
+
+      unless user
+        return Rails.logger.info "User #{user_id} not found"
+      end
+
+      ActiveRecord::Base.transaction do
+        unless user.phone_number_authentication
+          return Rails.logger.info "User #{user.id} does not have a phone number authentication"
+        end
+
+        user.facility_group.facilities.update_all(updated_at: Time.current)
+
+        if enable
+          Flipper.enable(:block_level_sync, user)
+        else
+          Flipper.disable(:block_level_sync, user)
+        end
+
+        Rails.logger.info "Block level sync #{enable ? "enabled" : "disabled"} for #{user.id}"
+      end
+    end
+  end
+
+  def bump(percentage)
+    existing_percentage = Flipper[:block_level_sync].percentage_of_actors_value
+    existing_enabled_user_ids = enabled_user_ids.to_set
+
+    ActiveRecord::Base.transaction do
+      Flipper.enable_percentage_of_actors(:block_level_sync, existing_percentage + percentage)
+
+      newly_enabled_user_ids = enabled_user_ids.to_set - existing_enabled_user_ids.to_set
+      newly_enabled_user_ids.each { |user_id| touch_facilities(User.find(user_id)) }
```

Comment:
Touching facilities one _user_ at a time seems inefficient. If two users from the same FG are enabled, we'd be touching the FG's facilities twice, right? Can we instead assemble the full flat unique list of facilities and then touch them?

Reply:
I don't particularly want to change this because I don't think it matters. It's a simple script that piggy-backs on an existing method that does it by user. I doubt we'll see any useful gains by doing this.

Reply:
That's a fair point. Keeping the code simple for a transient script like this makes sense 👍